content-src/components/DiscoveryStreamComponents/List/List.jsx

```diff
@@ -44,7 +45,10 @@ export class ListItem extends React.PureComponent {
           url={this.props.url}>
           <div className="ds-list-item-text">
             <div>
-              <div className="ds-list-item-title">{this.props.title}</div>
+              <div className="ds-list-item-title"
+                data-title-lines="2"
+                data-total-lines="4"
+                ref={clampTitleExcerpt}>{this.props.title}</div>
```

Comment:
It might be clearer to put the `ref` on a parent wrapper since you're implicitly operating on the next sibling as well, but not absolutely necessary.

Reply:
As in have the ref callback iterate through its children to find which ones to clamp?

Reply:
Yeah, something like putting `data-total-lines` on the wrapper and then `data-lines` on the elements themselves perhaps. Just something more explicitly targeted...

Reply:
That seems reasonable. I suppose from the parent we can `querySelectorAll("[data-clamp]")`. Wolasi wants to have at least 1 line of excerpt which we can bake into the API to count the number of things that want clamping and do some math. And the last thing / excerpt doesn't need an explicit number as it's the thing getting excess lines.